[ADD] partner_multi_relation_archive_propagate

ntsirintanis opened the pull request.

astirpe committed:
[13.0][ADD] partner_iterative_archive

oca-travis committed:
[UPD] Update partner_iterative_archive.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

newtratip committed:
[MIG] partner_iterative_archive: Migration to 14.0

oca-travis committed:
[UPD] Update partner_iterative_archive.pot

OCA-git-bot committed:
partner_iterative_archive 14.0.1.0.1

primes2h committed:
Added translation using Weblate (Italian)

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_iterative_archive
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_iterative_archive/it/

OCA-git-bot committed:
[UPD] README.rst

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (4 of 4 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_iterative_archive
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_iterative_archive/es/

ntsirintanis committed:
[UPD] partner_multi_relation_function: search function

hbrunn committed:
[ADD] partner_street_number

AungKoKoLin1997 committed:
[ADD] partner_address_format_domestic

imlopes committed:
[16.0][FIX] partner_company_default: avoid set company_id when create user

GeysonTrey committed:
[16.0][ADD] partner_quality_log: This module adds the “Qualification log” tab to the contact form

AungKoKoLin1997 committed:
[IMP] partner_contact_address_default: add option to allow selecting any partner

ntsirintanis committed:
[16.0][ADD] base_partner_merge_user_consolidation

marielejeune committed:
[IMP] partner_multi_relation: allow to archive relation types

Ryoko04 committed:
Added translation using Weblate (Japanese)

OCA-git-bot committed:
Merge PR #2187 into 16.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_contact_address_default.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-16.0/partner-contact-16.0-partner_contact_address_default
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_contact_address_default/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (13 of 13 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_contact_address_default
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_contact_address_default/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (20 of 20 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_industry_secondary
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_industry_secondary/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (41 of 41 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location_nuts
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location_nuts/it/

ntsirintanis commented:
@NL66278

Ryoko04 committed:
Translated using Weblate (Japanese)

Currently translated at 100.0% (13 of 13 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_contact_address_default
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_contact_address_default/ja/

OCA-git-bot committed:
Merge PR #2200 into 16.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_multi_relation.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-16.0/partner-contact-16.0-partner_multi_relation
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_multi_relation/

NL66278 commented on partner_multi_relation_archive_propagate/models/__init__.py:
```diff
@@ -0,0 +1,5 @@
+# Copyright 2025 Therp BV
```
No copyright lines in init files.

NL66278 commented on partner_multi_relation_archive_propagate/models/res_partner.py:
```diff
@@ -0,0 +1,91 @@
+                )
+        res = super().write(vals)
+        if "active" in vals:
+            if vals["active"] is False and archiving_companies:
```
You can leave out the test on vals, because otherwise there would be no archiving_companies. Same for the unarchiving.

NL66278 commented on partner_multi_relation_archive_propagate/models/res_partner.py:
```diff
@@ -0,0 +1,91 @@
+    def _archive_related_partners_via_relations(self):
+        """Archive related partners for organisational partners via relations."""
+        for partner in self:
+            if not partner.is_company:
```
You already checked for this.

OCA-git-bot committed:
Merge PR #2061 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update partner_address_format_domestic.pot

OCA-git-bot committed:
[BOT] post-merge updates

ntsirintanis committed:
[ADD] partner_multi_relation_archive_propagate

mymage committed:
Added translation using Weblate (Italian)

legalsylvain committed:
[FIX] partner_firstname: overload missing 'private' partner form view
The form view is used in HR > employee > Private Information at least.
add first and last names fields.

Ryoko04 committed:
Translated using Weblate (Japanese)

Currently translated at 100.0% (13 of 13 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_contact_address_default
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_contact_address_default/ja/

OCA-git-bot committed:
Merge PR #2191 into 16.0

Signed-off-by NL66278

Commit:
[UPD] Update base_partner_merge_user_consolidation.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2142 into 16.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_quality_log.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_partner_merge_user_consolidation
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_partner_merge_user_consolidation/it/

NL66278 committed:
[TST] pmr_function: add test for search

OCA-git-bot committed:
Merge PR #2249 into 16.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_multi_relation_function.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-16.0/partner-contact-16.0-partner_multi_relation_function
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_multi_relation_function/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (14 of 14 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_multi_relation_function
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_multi_relation_function/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_address_format_domestic
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_address_format_domestic/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_vat_unique
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_vat_unique/it/

uvid-josipa committed:
Added translation using Weblate (Croatian)

uvid-josipa committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (6 of 6 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_manual_rank
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_manual_rank/hr/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (28 of 28 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_quality_log
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_quality_log/it/

NL66278 commented:
@ntsirintanis I did not see the button to archive partner with children on the customer test instance. I realised that showing the button is computed in method `_compute_show_prop_wizard_button`. This must be overriden to also show the button when the partner is a company and has active connections.

NL66278 commented on partner_multi_relation_archive_propagate/models/res_partner.py:
```diff
@@ -0,0 +1,86 @@
+            )
+        )
+
+    @api.depends("child_ids", "is_company")
```
No need for @depends in this case. Field is not stored and is_company is unlikely to change on the fly...

NL66278 commented on partner_multi_relation_archive_propagate/models/res_partner.py:
```diff
@@ -0,0 +1,86 @@
+
+    @api.depends("child_ids", "is_company")
+    def _compute_show_prop_wizard_button(self):
+        # Keep original logic (children) and extend for relation-based propagation.
```
Although method overridden is trivial for now, still fully overriding a method is bad practice. Maybe the super method will change, or there will be other modules that want to force show of the button.

So why not do it this way:
```
for partner in partners:
    if partner.is_company and partner._has_propagating_relations():
        partner.show_prop_wizard_button = True
    else:
        super(ResPartner, partner)._compute_show_prop_wizard_button()
```

NL66278 commented:
@ntsirintanis Another change needed unfortunately. The button is now correctly shown for company partners with propagatable connections. However after hitting the button, the wizard is not shown, but the archiving starts immediately. Which works and also unarchiving works. But in fact we should first show the users the partners that will be archived, which means also enhancing the `_get_descendants` method to not only return contact children, but also connected partners.

ntsirintanis commented:
> @ntsirintanis Another change needed unfortunately. The button is now correctly shown for company partners with propagatable connections. However after hitting the button, the wizard is not shown, but the archiving starts immediately. Which works and also unarchiving works. But in fact we should first show the users the partners that will be archived, which means also enhancing the `_get_descendants` method to not only return contact children, but also connected partners.

hm yes, I think this will be better if we actually override action_archive_with_contacts, to return a wizard or not based on conditions (company, related partners)

ntsirintanis commented on partner_multi_relation_archive_propagate/models/res_partner.py:
```diff
@@ -0,0 +1,124 @@
+            )
+            company._notify_skipped_partners(unarchivable)
+
+    def action_archive_with_contacts(self):
```
this  takes care to open a wizard for companies with relations

NL66278 replied:
The problem, apart from significant code overlap, is that in the wizard we might need to show both children (in the normal Odoo sense), and descendants (through relations). In short all partners that are to be archived as a result of propagation. What was your objection to overriding `_get_desendants` to return the union of both?

ntsirintanis replied:
No need to touch _get_descendants at all with this modification, as this is the very last step before wizard is opened or partners are archived automatically. I do not want to touch _get_descendants anyway because it’s used by _archive_propagate_wizard() / _archive_propagate_external() and it logically returns children, not any other relation .

ntsirintanis replied:
I will see however into introducing a new method in the base module, partner_archive_propagate, that simply returns the result of _get_descendants, and override that in this module. We should achieve what I proposed and what you proposed in one shot

NL66278 replied:
That seems like a great idea.

ntsirintanis committed:
[UPD] partner_archive_propagate: new helper method for descendants

ntsirintanis committed:
[ADD] partner_multi_relation_archive_propagate: get all archivable partners in wizard

luisDIXMIT committed:
[IMP] partner_iterative_archive: pre-commit auto fixes

luisDIXMIT committed:
[MIG] partner_iterative_archive: migration to 16.0

NL66278 reviewed (approved):
👍 And succesfully tested

OCA-git-bot committed:
Merge PR #2231 into 16.0

Signed-off-by mmequignon

OCA-git-bot committed:
[BOT] post-merge updates

baguenth committed:
Translated using Weblate (German)

Currently translated at 100.0% (13 of 13 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_contact_address_default
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_contact_address_default/de/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 46.1% (12 of 26 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location_geonames_import
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location_geonames_import/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 88.5% (31 of 35 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_external_map
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_external_map/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 81.6% (40 of 49 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_identification/sl/

risto42 committed:
Added translation using Weblate (French)

risto42 committed:
Translated using Weblate (French)

Currently translated at 100.0% (27 of 27 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_stage
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_stage/fr/

risto42 committed:
Added translation using Weblate (French)

risto42 committed:
Translated using Weblate (French)

Currently translated at 100.0% (18 of 18 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_tier_validation/fr/

NL66278 committed:
[UPD] Update CI files through copier update --UNSAFE

NL66278 committed:
[FIX] pc_age_range: remove contraproductie freeze_time

test setup creates partners with birthdate_date related to current date, tests uses freeze_gun, leading to partners with
a negative age (\!), and then not fitting in the expected baby date range.

OCA-git-bot committed:
Merge PR #2301 into 16.0

Signed-off-by NL66278

OCA-git-bot committed:
Merge PR #2258 into 16.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_iterative_archive.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #2027 into 16.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_street_number.pot

OCA-git-bot committed:
[BOT] post-merge updates

eugenichafer committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (4 of 4 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_partner_company_group
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_partner_company_group/ca/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (38 of 38 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_street_number
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_street_number/it/

risto42 committed:
Translated using Weblate (French)

Currently translated at 100.0% (22 of 22 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_contact_department
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_contact_department/fr/

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_contact_access_link
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_contact_access_link/it/

OCA-git-bot committed:
Merge PR #2106 into 16.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_company_default.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-16.0/partner-contact-16.0-partner_company_default
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_company_default/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_company_default
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_company_default/it/

OCA-git-bot committed:
Merge PR #2212 into 16.0

Signed-off-by NL66278

Commit:
[UPD] Update partner_archive_propagate.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

ntsirintanis committed:
[FIX] partner_archive_propagate: actually do not archive deleted lines from wizard
